src/wasi/Wasi.cpp

```diff
@@ -108,12 +109,92 @@ WASI::WASI()
     init_options.argc = 0;
     init_options.argv = nullptr;
     init_options.envp = nullptr;
-    init_options.preopenc = 0;
+    if (shareHostEnv) {
+        std::vector<std::string> variables = {
+            // Common Unix environment variables.
+            "PATH",
+            "LIB",
+            "PWD",
+            "SHELL",
+            "TERM",
+            "HOME",
+            "LOGNAME",
+            "HOSTNAME",
+            "UID",
+            "TEMP",
+            "EDITOR"
+            "LANG",
+            "LC_TIME",
+            // Common Windows environment variables.
+            "APPDATA",
+            "USERPROFILE",
+            "USERDOMAIN",
+            "windir",
+            "ProgramFiles",
+            "OS"
+            "LOCALAPPDATA",
+            "SystemRoot",
+            "SystemDrive",
+        };
+
+        char** envp = (char**)malloc(sizeof(char**) * variables.size());
```

Comment:
What about simply allocating a vector of `std::string` here?
This kind of manual memory alloc is so complicated to handle and hard for correct freeing

Reply:
After spending some time thinking how I would change it to be easily used with uvwasi, I could not think of a way. If you have an idea, it would be greatly appriciated.

Reply:
I see. I'll revise this later